.github/workflows/test_and_make_release.yml

```diff
@@ -12,24 +12,131 @@ on:
       - master
 
 jobs:
-  test:
+  template_test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: [3.8, 3.7, 3.6]
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Install tools (Poetry)
+        uses: Gr1N/setup-poetry@v2
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          pip install tox
+          pip install cookiecutter
+      - name: Instantiate template and validate
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
+        run: |
+          cookiecutter --default-config --no-input .
+          cd python_boilerplate
+          make test
```

Comment:
does it need `make setup-dev-env-minimal` first ?

Reply:
Actually, it should run an equivalent of `make setup-cicd-test-stage`, which was initially for gitlab-ci. That should be run in the previous step, which currently does the needed `pip install tox`. 

`pip install cookiecutter` should however stay outside the project's `Makefile`, since it is not part of the python project ifself.

Reply:
DONE. I further split the jobs. have a look and let me know.